akka-typed/src/main/scala/akka/typed/internal/DeathWatch.scala

```diff
@@ -70,13 +80,18 @@ private[typed] trait DeathWatch[T] {
    */
   protected def watchedActorTerminated(actor: ARImpl, failure: Throwable): Boolean = {
     removeChild(actor)
-    if (watching.contains(actor)) {
+    watching.get(actor).foreach { optionalMessage ⇒
```

Comment:
Aah so the `Function1` object that gets passed to `foreach` gets allocated each time - I never realized that, good one!